sdk/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -826,4 +828,12 @@ protected boolean inProject(@NonNull String className) {
     protected Collection<BeforeRecordBreadcrumb> getBeforeRecordBreadcrumbTasks() {
         return beforeRecordBreadcrumbTasks;
     }
+
+    void addBeforeSendSession(BeforeSendSession beforeSendSession) {
+        sessionCallbacks.add(beforeSendSession);
+    }
+
+    Collection<BeforeSendSession> getSessionCallbacks() {
```

Comment:
is it possible to return an iterator that represents this list? The fact you can get hold of the collection and mutate it means its privacy is leaked and it may as well just be a public field rather than a private one with a getter?

Reply:
It would be possible to wrap this with something like `Collections.unmodifiableList`, or even convert the class to Kotlin and use truly immutable collections. Other [similar callbacks](https://github.com/bugsnag/bugsnag-android/blob/6f6c88b1a416e02f0bf3fea0cdf3d5cd34fabc4c/sdk/src/main/java/com/bugsnag/android/Configuration.java#L494) don't seem to do this though, so for now I'd lean in favour of staying consistent with that implementation.